src/main_window.cpp

```diff
@@ -264,19 +264,51 @@ void MainWindow::installPkg()
     // Create game directory.
     auto gamesDirectory = readGamesDirectorySetting();
 
-    if (!QDir(gamesDirectory).mkdir(param.titleId())) {
-        QString msg(
-            "Cannot create directory %1 inside %2. "
-            "If you have a failed installation from a previous attempt, you will need to remove this directory before trying again.");
+    // Get Param information
+    auto appver = param.appver();
+    auto category = param.category();
+    auto shortContentId = param.shortContentId();
+    auto title = param.title();
+    auto titleId = param.titleId();
+
+    // Check if file is Patch/DLC.
+    bool patchOrDlc = false;
+    if (category.startsWith("gp") || category.contains("ac")) {
+        patchOrDlc = true;
+    }
 
-        QMessageBox::critical(&progress, "Error", msg.arg(param.titleId()).arg(gamesDirectory));
-        return;
+    // If the game exists and the file isn't a Patch/DLC, don't allow.
+    if (!QDir(gamesDirectory).mkdir(titleId)) {
+        if (patchOrDlc || category.startsWith("gp")) {
+            QString msg("Install directory for %1 could not be created at %2.");
+
+            QMessageBox::critical(&progress, "Cannot create install directory", msg.arg(titleId).arg(gamesDirectory));
+            return;
+        } else {
+            QString msg("PKG file cannot be installed as it is not a patch or DLC for preexisting application %1 at %2.");
+
+            QMessageBox::critical(&progress, "Invalid PKG file. (Not Patch/DLC for Existing Game)", msg.arg(titleId).arg(gamesDirectory));
```

Comment:
I think the message box title is too long. It should be the category of the message like `Error` in case of the error.